components/eamxx/src/control/intensive_observation_period.hpp

```diff
@@ -86,10 +135,66 @@ class IntensiveObservationPeriod
     int local_column_index_of_closest_column;
   };
 
+  // Struct for storing relevant time information
+  struct TimeInfo {
+    util::TimeStamp iop_file_begin_time;
+    view_1d_host<int> iop_file_times_in_sec;
+
+    int time_idx_of_current_data = -1;
+
+    int get_iop_file_time_idx (const util::TimeStamp& current_ts)
+    {
+      // Get iop file time index that the given timestamp falls between.
+      // Note: the last time in iop file represents the non-inclusive
+      //       upper bound of acceptable model times.
+      const auto n_iop_times = iop_file_times_in_sec.extent(0);
+      int time_idx=-1;
+      for (size_t t=0; t<n_iop_times-1; ++t) {
+        if (iop_file_begin_time + iop_file_times_in_sec(t) <= current_ts &&
+            current_ts < iop_file_begin_time + iop_file_times_in_sec(t+1)) {
+          time_idx = t;
+        }
+      }
+
+      EKAT_REQUIRE_MSG(time_idx>=0,
+                       "Error! Current model time ("+current_ts.to_string()+") is not within "
+                       "IOP time period: ["+iop_file_begin_time.to_string()+", "+
+                       (iop_file_begin_time+iop_file_times_in_sec(n_iop_times-1)).to_string()+").\n");
+      return time_idx;
+    }
+  };
```

Comment:
Looks like this could be used in the eamxx_time_interpolation.*pp utilities...

Reply:
I'll take a look.

Reply:
No need to do it here. Just saying, if you think it's usable there, we may open a code-cleanup issue. @AaronDonahue may also know whether it could be used there too.